pkg/controller/controller_instance.go

```diff
@@ -1629,11 +1694,13 @@ func (c *controller) processUpdateServiceInstanceAsyncResponse(instance *v1beta1
 // processDeprovisionSuccess handles the logging and updating of
 // a ServiceInstance that has successfully been deprovisioned at the broker.
 func (c *controller) processDeprovisionSuccess(instance *v1beta1.ServiceInstance) error {
-	mitigatingOrphan := instance.Status.OrphanMitigationInProgress
+	mitigatingOrphan := isServiceInstanceOrphanMitigation(instance)
 
 	reason := successDeprovisionReason
 	msg := successDeprovisionMessage
 	if mitigatingOrphan {
+		removeServiceInstanceCondition(instance, v1beta1.ServiceInstanceConditionOrphanMitigation)
```

Comment:
Important change 2: Explicitly reset the orphan mitigation condition (and flag) after we have successfully finished orphan mitigation.